### PR TITLE
change channel order according to cfg

### PR DESCRIPTION
**Pull request by liuxin9608**

### CODE REVIEW

File: demo/top_down_video_demo_with_mmdet.py [hunk 1]
```diff
@@ -83,7 +84,10 @@ def main():
 
     dataset = pose_model.cfg.data['test']['type']
 
-    cap = cv2.VideoCapture(args.video_path)
+    if args.video_path != '':
+        cap = cv2.VideoCapture(args.video_path)
+    else:
+        cap = cv2.VideoCapture(0)
```
Comment (innerlee) [inline demo/top_down_video_demo_with_mmdet.py:90]: Is this for live camera?

Comment (liuxin9608) [reply demo/top_down_video_demo_with_mmdet.py:90]: yes

Comment (innerlee) [reply demo/top_down_video_demo_with_mmdet.py:90]: Refer to https://github.com/open-mmlab/mmaction2/blob/master/demo/webcam_demo.py for an implementation of webcam. It uses two threads for capturing and analyzing, which improves user experience

Comment (innerlee) [reply demo/top_down_video_demo_with_mmdet.py:90]: So I suggest writing a separate file for webcam, and make this pr focus on channel order

File: mmpose/apis/inference.py [hunk 2]
```diff
@@ -117,18 +118,23 @@ def __call__(self, results):
...
         if isinstance(results['img_or_path'], str):
             results['image_file'] = results['img_or_path']
-        else:
+            img = mmcv.imread(results['img_or_path'], self.color_type,
+                              self.channel_order)
+        elif isinstance(results['img_or_path'], np.ndarray):
             results['image_file'] = ''
-        img = mmcv.imread(results['img_or_path'], self.color_type,
-                          self.channel_order)
+            if self.channel_order == 'rgb':
```
Comment (innerlee) [inline mmpose/apis/inference.py:132]: add check for color_type for this if statement

File: demo/top_down_video_demo_with_mmdet.py (hunk 1 above)
Comment (jin-s13) [reply demo/top_down_video_demo_with_mmdet.py:90]: agree

**Review by innerlee — APPROVED** [review]

**Pull request merged**